File: examples/faker/company.test.js
import { describe, expect } from "https://jslib.k6.io/k6chaijs/4.3.4.3/index.js";
import { Faker } from "k6/x/faker";

export default function () {
  describe("company", (t) => {
    let faker = new Faker(11);
    expect(faker.company.name(), "name()").to.equal("Xatori");
    faker = new Faker(11);
    expect(faker.company.suffix(), "suffix()").to.equal("LLC");
    faker = new Faker(11);
    expect(faker.company.buzzWord(), "buzzWord()").to.equal("Reverse-engineered");
    faker = new Faker(11);
    expect(faker.company.bs(), "bs()").to.equal("24-7");
  });
}

File: examples/faker/company_test.go
package faker

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func company(t *testing.T) goCompany {
	t.Helper()

	return &goCompanyImpl{fakeit: fakeit(t)}
}

func Test_goCompany(t *testing.T) {
	t.Parallel()
	must := noError(t)

	assert.Equal(t, "Xatori", must(company(t).nameMethod()))
	assert.Equal(t, "LLC", must(company(t).suffixMethod()))
	assert.Equal(t, "Reverse-engineered", must(company(t).buzzWordMethod()))
	assert.Equal(t, "24-7", must(company(t).bsMethod()))
}
